src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -46,97 +45,84 @@
 
-    private void processZipFiles(CloudBlobContainer container) {
-        log.info("Processing blobs for container {} ", container.getName());
+    private void processZipFiles(CloudBlobContainer container)
+        throws IOException, StorageException, URISyntaxException {
+        log.info("Processing blobs for container {}", container.getName());
 
         for (ListBlobItem blobItem : container.listBlobs()) {
             String zipFilename = FilenameUtils.getName(blobItem.getUri().toString());
 
-            try {
-                processZipFile(container, zipFilename);
-            } catch (Exception e) {
-                //If any error occurs processing one record remaining records should be continued processing
-                log.error("Exception occurred while processing zip file " + zipFilename, e);
-                // TODO A record in Database needs to created with appropriate status.
-            }
+            processZipFile(container, zipFilename);
         }
     }
 
     private void processZipFile(CloudBlobContainer container, String zipFilename)
-        throws StorageException, URISyntaxException, IOException {
+        throws IOException, StorageException, URISyntaxException {
 
-        List<Pdf> pdfFiles = new ArrayList<>();
         CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(zipFilename);
         BlobInputStream blobInputStream = cloudBlockBlob.openInputStream();
-        boolean isUploadFailure = false;
-        Envelope envelope = null;
 
         //Zip file will include metadata.json and collection of pdf documents
         try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
-            byte[] metadataStream = null;
-            ZipEntry zipEntry;
-            while ((zipEntry = zis.getNextEntry()) != null) {
-                switch (FilenameUtils.getExtension(zipEntry.getName())) {
-                    case "json":
-                        metadataStream = toByteArray(zis);
-                        break;
-                    case "pdf":
-                        Pdf pdf = new Pdf(zipEntry.getName(), toByteArray(zis));
-                        pdfFiles.add(pdf);
-                        break;
-                    default:
-                        //Contract breakage
-                        throw new NoPdfFileFoundException(
-                            "Zip file contains non pdf documents for file " + zipFilename
-                        );
-                }
-            }
+            Map<Envelope, List<Pdf>> envelopeMap = processZipFileEntry(zis, zipFilename, container.getName());
 
-            envelope = envelopeProcessor.processEnvelope(metadataStream, container.getName());
-            isUploadFailure = true;
-
-            documentProcessor.processPdfFiles(pdfFiles, envelope.getScannableItems());
-            envelopeProcessor.markAsUploaded(envelope, container.getName(), zipFilename);
+            for (Map.Entry<Envelope, List<Pdf>> entry : envelopeMap.entrySet()) {
```

Comment:
There will be always one envelope and collection of pdf's. Any reason why we are using map here and then iterating through entry set ?

Reply:
Also you have  a new class now processing of zip files can it be moved to that class ?

Reply:
Good point. Will simplify error wrapper